File: app/models/feedback.rb
# frozen_string_literal: true

class Feedback < ApplicationRecord
  enum status: %i[active inactive]

  has_many :users_feedbacks
  has_many :users, through: :users_feedbacks
  has_many :reviewers, through: :users_feedbacks, class_name: 'User', foreign_key: 'reviewer_id'
  has_many :feedbacks_questions
  has_many :questions, through: :feedbacks_questions

  validates :title, presence: true
  validates :status, presence: true, inclusion: { in: Feedback.statuses }
  USE_COLUMN_NAMES = %i[id title status].freeze
end

File: app/javascript/packs/employee/models/Feedback.ts
import User from "./User";
import Question from "./Question";
import FeedbacksQuestion from "./FeedbacksQuestion";

class Feedback {
  id: number;
  title: string;
  status: string;
  questions?: Array<Question>;
  feedbacksQuestions?: Array<FeedbacksQuestion>;
  usersFeedbacks: Array<{user: User, reviewer: User}>;

  constructor(data: {
    id: number;
    title: string;
    status: string;
    questions?: Array<Question>;
    feedbacksQuestions?: Array<FeedbacksQuestion>;
    usersFeedbacks: Array<{user: User, reviewer: User}>;
  }) {
    this.id = data.id;
    this.title = data.title;
    this.status = data.status;
    this.questions = data.questions;
    this.feedbacksQuestions = data.feedbacksQuestions;
    this.usersFeedbacks = data.usersFeedbacks;
  }
}

export default Feedback;
